src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -30,6 +30,8 @@
 
 use Doctrine\ORM\EntityManagerInterface;
 use Exception;
+use HelperTreeCategories;
```

Comment:
Hi @Codencode 

PHPStan blocks the use of legacy classes in recent code by default, it's a rule to encourage us not to rely on legacy too much and develop new components in favor of old ones. Vut it's not always applicable so we pick the exceptions one by one and this is one of them.

So to allow this modification we need to update the baselines that lists all these exceptions, you can do so by running this command `php -d memory_limit=-1 ./vendor/bin/phpstan analyse -c phpstan.neon.dist --generate-baseline`

Reply:
Hi @jolelievre,

While running the PHPStan analysis, the system wasn't adding the exceptions correctly for me. I probably have something not properly configured in my local environment, so I went ahead and added the exceptions manually in the file, and it worked.

Thanks a lot for the tip!!!